tests/integrations/test_messagingqueues.py

```diff
@@ -0,0 +1,90 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2023 Datadog, Inc.
+
+from utils import weblog, interfaces, scenarios
+from utils.tools import logger
+
+
+@scenarios.integrations
+class Test_MQKafka:
+    def setup_apm_kafka_default_context_propagation(self):
+        self.r = weblog.get("/dsm?integration=kafka")
+
+    def setup_dsm_kafka_default_context_propagation(self):
+        self.r = weblog.get("/dsm?integration=kafka")
+
+    def test_dsm_kafka_default_context_propagation(self):
+        for data in interfaces.agent.get_dsm_data():
+            for stats_bucket in data["request"]["content"].get("Stats", {}):
+
+                producer_stats_point = None
+                consumer_stats_point = None
+
+                for stats_point in stats_bucket.get("Stats", {}):
+                    if stats_point["EdgeTags"][0] == "direction:in":
+                        consumer_stats_point = stats_point
+                    elif stats_point["EdgeTags"][0] == "direction:out":
+                        producer_stats_point = stats_point
+
+                # consumers are a direct child of a producer span
+                assert consumer_stats_point["ParentHash"] == producer_stats_point["Hash"]
+
+    # kafka in dd-trace-py does not propagate span context for tracing purposes
+    def test_apm_kafka_default_context_propagation(self):
+        spans_payload = interfaces.library.get_spans(self.r)
+
+        # get_spans returns several tuple items
+        for items in spans_payload:
+            # each iteration of spans_payload is a tuple
+            data, trace, root_span = items
+            producer_span = None
+            consumer_span = None
+
+            for span in trace:
+                span_component = span["meta"]["component"]
+                if "span.kind" in span["meta"].keys():
+                    span_kind = span["meta"]["span.kind"]
+                else:
+                    span_kind = "none"
+
+                if span_component == "kafka" and span_kind == "producer":
+                    logger.debug("producer span found:", span)
+                    producer_span = span
+                elif span_component == "kafka" and span_kind == "consumer":
+                    logger.debug("consumer span found:", span)
+                    consumer_span = span
+
+            assert producer_span is not None
+            assert consumer_span is not None
+
+            # consumer and producer spans have no direct parent/child relationship
+            assert consumer_span["parent_id"] != producer_span["span_id"]
```

Comment:
I'd remove these, we don't want to assert current behavior, we want to assert desired behavior

Reply:
That's fair, I flipped the logic of these tests back to what they should be!

Reply:
Marking this as resolved as https://github.com/DataDog/system-tests/pull/1783 will cover assertions for APM tracing.